acceptance/suites/tests/020_regression_tests/SERVER-571_hiera_backends_in_modules.rb

```diff
@@ -0,0 +1,53 @@
+test_name "(SERVER-571) Test hiera backend plugins located in modules"
+
+# The way this "works" is that the user installs a module on the server, the
+# agent run synchronizes the module lib/ plugins to the [Puppet Installation
+# Layout][layout] `libdir` path of /opt/puppetlabs/puppet/cache/lib  The
+# master then loads the synchronized plugin file using a ruby `require` which
+# searches the $LOAD_PATH
+#
+# [layout]: http://goo.gl/BWyXpo
+
+this_file = __FILE__
+# The local directory where resources are located, e.g. an environment directory
+rsrc_dir = File.expand_path('../../../../resources', this_file)
+
+# The target directory to place whole environments.
+# e.g. /etc/puppetlabs/code/environments
+env_path = on(master, puppet("config print environmentpath")).stdout.chomp
+target_dir = env_path.split(master['pathseparator']).first
+
+scp_options = {}
+step "Install hiera_test module in server571 environment on the master"
+Dir.chdir("#{rsrc_dir}/environments") do
+  # Create /etc/puppetlabs/code/environments/server571
+  # so that agents can run against this environment to sync plugins
+  master.do_scp_to "server571", target_dir, scp_options
+end
+
+# Save the hiera configuration
+step "Save hiera_config file to restore in teardown"
+hiera_config = on(master, puppet('config print hiera_config')).stdout.chomp
+on(master, "cp -p #{hiera_config} #{hiera_config}.bak")
+
+step "Write hiera configuration that uses the backend from the module"
+Dir.chdir("#{rsrc_dir}/files/server571") do
+  master.do_scp_to "hiera.yaml", hiera_config, scp_options
+end
+
+teardown do
+  on(master, "mv -f #{hiera_config}.bak #{hiera_config}")
```

Comment:
Pardon my Beaker ignorance...

If the test fails before original hiera_config file is backed up, will this teardown step still run, fail, and the ultimate result of the test failure appear to be due to the "mv" failure as opposed to the initial failure?